feat(TX-295): ACH payment page redesign

rquartararo opened the pull request:
The type of this PR is: Feature
<!-- Bugfix/Feature/Enhancement/Documentation -->

<!-- If applicable, write the Jira ticket number in square brackets e.g. `[GRO-434]`
     The Jira integration will turn it into a clickable link for you. -->

This PR solves [TX-295]

### Description

Updates payment transaction step design to remove the two-step process in favor of a more simple implementation.  (behind stripe_ACH feature flag)

This does not include some of the smaller details of the Figma design, as I want to get the new logic out first as it could affect future development. 

### Recording 

https://user-images.githubusercontent.com/50849237/165928682-421819c9-1ba9-4300-97b5-d35d9736b5cd.mov




[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TX-295]: https://artsyproduct.atlassian.net/browse/TX-295?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

rquartararo committed:
feat: new payment page design

rquartararo committed:
test: update tests for new design

rquartararo commented:
I'm running into this error Invariant Violation: commitMutation: expected "environment" to be an instance of "RelayModernEnvironment" when running the ACH tests and I'm having a hard time figuring out what could be causing this.

erikdstock commented:
By mocking the BankDebitForm in [this commit](https://github.com/artsy/force/pull/10003/commits/b7255f76e186be6cc2b99918caa2117d8a4dcbf1) i was able to fix some of the failing tests - currently-commented out code is because i copied it from another file.

starsirius committed:
refactor: move test env setup to a helper

starsirius committed:
test: mock BankDebitProvider

h/t @erikdstock

The ACH tests are raising an error:

```
Invariant Violation: commitMutation: expected "environment" to be an instance of "RelayModernEnvironment"
```

Mocking the BankDebitProvider makes them pass again, and we'll revisit
how to properly test the Metaphysics mutation for creating a bank debit
setup.

starsirius committed:
test: disable jest/expect-expect

starsirius committed:
test: check collapse property instead of presence of child element which should always be present

starsirius committed:
test: add a todo test

starsirius commented:
Thanks, @erikdstock! 🙏  I pulled in the mock in this PR, fixed some tests and added a todo for testing the mutation (which I can't make it work). It's 🍏  ! I'm going to merge it to unblock other work streams!

starsirius commented on src/v2/Apps/Order/Routes/__tests__/Payment.jest.tsx:
```diff
@@ -208,23 +234,34 @@ describe("Payment", () => {
-      const page = await buildPage()
-      page.selectPaymentMethod(1)
-      expect(page.text()).toContain("Bank transfer")
-      expect(page.find(BankDebitProvider).length).toBe(1)
```
Checking for presence doesn't work in this case because we are using `Collapse` and it's just changing the element heights. Updated below to check the props on the corresponding `Collapse` component.
